src/components/Button/index.tsx

```diff
@@ -99,24 +108,70 @@ const buttonVariants = cva(
   }
 )
 
+const buttonVariants = cva(
+  [
+    'inline-flex',
+    'items-center',
+    'justify-center',
+    'transition-colors',
+    'focus-visible:outline-none',
+    'focus-visible:ring-1',
+    'focus-visible:ring-ring',
+    'bg-transparent',
+    'disabled:pointer-events-none',
+    'pr-4',
+    'pl-4',
+
+  ]
+)
+
+const selectedVariant = cva(
+  [
+    'border-violet-700'
+  ]
+)
+
 export interface ButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
-    VariantProps<typeof buttonVariants> {
-  asChild?: boolean
+    VariantProps<typeof formVariants> {
+  leftSideChild?: React.ReactNode
+  rightSideChild?: React.ReactNode,
+  leftSideClassName?: string,
+  rightSideClassName?: string
+  asChild?: boolean,
+  selected?:boolean
 }
 
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  ({ className, variant, size, asChild = false, ...props }, ref) => {
+  ({ className, variant, size, asChild = false,leftSideChild, rightSideChild, leftSideClassName, rightSideClassName,disabled=false, selected=false, ...props }, ref) => {
+
+    const [isSelected, setIsSelected] = React.useState(selected)
+
     const Comp = asChild ? Slot : 'button'
     return (
-      <Comp
-        className={cn(buttonVariants({ variant, size, className }))}
-        ref={ref}
-        {...props}
-      />
+        <form className={cn(formVariants({ variant }), className, isSelected? selectedVariant():null)} disabled={disabled} onClick={()=> setIsSelected(!isSelected)}>
+          {leftSideChild?
+            <div className={cn(leftSideVariants(), leftSideClassName)}>
+              {leftSideChild}
+            </div>:<></>
+          }
+          <button
+            className={cn(buttonVariants(), className)}
+            ref={ref}
+            {...props}
+          />
+          {rightSideChild?
+          
+          <div className={cn(rightSideVariants(), rightSideVariants)}>
+             {rightSideChild}
+          </div>
+           :<></>
+          }
+      </form>
```

Comment:
**suggestion:** You don't need a `form` element wrapping the button. A `button` element accepts arbitrary `children`, so I would:
- remove the `form`
- move the `leftSideChild` and `rightSideChild` inside the `<button>` 
Something like:
```suggestion
        <button
          className={cn(buttonVariants(), className)}
          ref={ref}
          {...props}
        >
          {leftSideChild ? (
            <div className={cn(leftSideVariants(), leftSideClassName)}>
              {leftSideChild}
            </div>
          ) : (
            <></>
          )}
          {children}
          {rightSideChild ? (
            <div className={cn(rightSideVariants(), rightSideVariants)}>
              {rightSideChild}
            </div>
          ) : (
            <></>
          )}
        </button>
```